artifacts_test.py

```diff
@@ -436,7 +436,7 @@ def get_email_data(self):
         email_data = self._get_common_email_data()
         try:
             node = self.node
-        except Exception:  # pylint: disable=broad-except
+        except (ValueError, IndexError):
```

Comment:
It may be also `AttributeError` and some other.
I don't see any benefit narrowing down the exception coverage here.

Reply:
The `AttributeError` is totally possible here.

Reply:
yeah, if you remove `db_cluster` assignment from `Tester.__init__`

Reply:
The absence of the `nodes` attr is also one of possible causes.
So, the statement `those are the only two options` is false because there are 3 different possible exceptions.

Reply:
same as before it's in `nodes` are being assigned in `BaseCluster.__init__` 

to sum it up, we know exactly what can get broken here, i.e. the list of node can be empty, cause test failed before starting the node.